circuit-construction/src/writer.rs

```diff
@@ -577,9 +672,11 @@ pub trait Cs<F: FftField + PrimeField> {
                     }
                 });
 
+                // Accumulated chunk value
                 n = self.var(|| n.val().double().double() + crumb.val());
             }
 
+            // In final row, this is the input value, otherwise the accumulated value
             row[1] = if i == num_rows - 1 { x } else { n };
```

Comment:
Specially here, I think we should have an additional assert_eq between the intermediate values, and eventually, obtain an equality between the input value `x` and the final computed `n`.